### PR TITLE
hex2buf now returns an error in case of an odd number of characters

### PR DESCRIPTION
**Pull request by e-asphyx**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have run the linter against the changes
- [ ] You have added unit tests (if relevant/appropriate)
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [ ] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [ ] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

**Comment by dsawali** [comment]
Thank you for the prompt response @e-asphyx, could you adjust the unit tests to pass with the new change in hex2buf?

File: packages/taquito-utils/src/taquito-utils.ts [hunk 1]
```diff
@@ -197,12 +197,20 @@ export function encodeKeyHash(value: string) {
  * @throws {@link ValueConversionError}
  */
 export const hex2buf = (hex: string): Uint8Array => {
-  const match = hex.match(/[\da-f]{2}/gi);
-  if (match) {
-    return new Uint8Array(match.map((h) => parseInt(h, 16)));
-  } else {
+  if ((hex.length & 1) !== 0) {
```
Comment (AlirezaHaghshenas) [inline packages/taquito-utils/src/taquito-utils.ts:200]:
```suggestion
  if ((hex.length % 2) !== 0) {
```
It's more familiar to js developers

File: packages/taquito-utils/src/taquito-utils.ts [hunk 2]
```diff
@@ -197,12 +197,20 @@ export function encodeKeyHash(value: string) {
...
+    // odd length
     throw new ValueConversionError(hex, 'Uint8Array');
   }
+  const out = new Uint8Array(hex.length / 2);
+  let j = 0;
+  for (let i = 0; i < hex.length; i += 2) {
+    const v = parseInt(hex.slice(i, i + 2), 16);
```
Comment (AlirezaHaghshenas) [inline packages/taquito-utils/src/taquito-utils.ts:207]:
```suggestion
  for (let i = 0; i < hex.length; i += 1) {
    const v = parseInt(hex.charAt(i), 16);
```

If the second char is out of hex code, the `parseInt` function will only convert the first part, it does not return `nan`

Comment (AlirezaHaghshenas) [reply packages/taquito-utils/src/taquito-utils.ts:207]: Sorry, more changes are needed

File: packages/taquito-utils/src/taquito-utils.ts (hunk 1 above)
Comment (e-asphyx) [reply packages/taquito-utils/src/taquito-utils.ts:200]: Looks good

**Review by hui-an-yang — APPROVED** [review]
lgtm 🎉

**Pull request merged**